src/Riak/Api/Http.php

```diff
@@ -718,30 +719,27 @@ protected function parseResponse()
                 break;
 
             case 'Basho\Riak\Command\MapReduce\Fetch':
-                $results = null;
-                if ($body) {
-                    $results = json_decode($body);
-                }
+                $results = in_array($this->statusCode, [200,204]) ? json_decode($body) : null;
                 $response = new Command\MapReduce\Response($this->success, $this->statusCode, $this->error, $results);
                 break;
             case 'Basho\Riak\Command\Indexes\Query':
-                $body = json_decode(rawurldecode($body), true);
+                $json_object = in_array($this->statusCode, [200,204]) ? json_decode($body, true) : null;
                 $results = [];
                 $termsReturned = false;
                 $continuation = null;
                 $done = true;
 
-                if (isset($body['keys'])) {
-                    $results = $body['keys'];
+                if (isset($json_object['keys'])) {
```

Comment:
@lukebakken no it won't, php is *cough* clever *cough* in these cases, however I generally prefer to be on side of caution and I tend to always verify if the result of `json_decode` is an object or not. Also probably it's not a good idea to call the variable `$json_object` when in reality it's an associative array.